src/main/java/de/tum/in/www1/artemis/service/exam/StudentExamService.java

```diff
@@ -257,10 +257,18 @@ else if (submittedAnswer instanceof ShortAnswerSubmittedAnswer) {
                             quizSubmissionRepository.save((QuizSubmission) submission);
                         }
                         else if (exercise instanceof TextExercise) {
-                            textSubmissionRepository.save((TextSubmission) submission);
+                            TextSubmission existingSubmission = (TextSubmission) existingParticipation.findLatestSubmission().get();
+                            TextSubmission textSubmission = (TextSubmission) submission;
+                            if (!textSubmission.getText().equals(existingSubmission.getText())) {
+                                textSubmissionRepository.save(textSubmission);
+                            }
                         }
                         else if (exercise instanceof ModelingExercise) {
-                            modelingSubmissionRepository.save((ModelingSubmission) submission);
+                            ModelingSubmission existingSubmission = (ModelingSubmission) existingParticipation.findLatestSubmission().get();
+                            ModelingSubmission modelingSubmission = (ModelingSubmission) submission;
+                            if (!modelingSubmission.getModel().equals(existingSubmission.getModel())) {
```

Comment:
Is `getModel()` nullable (e.g. in case of empty submissions) ?

Reply:
Yes, it is. Thanks for pointing it out!